paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSheetViewModelTest.kt

```diff
@@ -3204,6 +3205,48 @@ class CustomerSheetViewModelTest {
         verify(eventReporter).onCardNumberCompleted()
     }
 
+    @Test
+    fun `When setting up with intent, should call 'IntentConfirmationInterceptor' with expected params`() =
+        runTest(testDispatcher) {
+            val intentConfirmationInterceptor = FakeIntentConfirmationInterceptor()
+
+            val viewModel = createViewModel(
+                workContext = testDispatcher,
+                initialBackStack = listOf(
+                    selectPaymentMethodViewState,
+                    addPaymentMethodViewState,
+                ),
+                stripeRepository = FakeStripeRepository(
+                    createPaymentMethodResult = Result.success(CARD_PAYMENT_METHOD),
+                    retrieveSetupIntent = Result.success(SetupIntentFixtures.SI_SUCCEEDED),
+                ),
+                customerAdapter = FakeCustomerAdapter(
+                    onSetupIntentClientSecretForCustomerAttach = {
+                        CustomerAdapter.Result.success("seti_123")
+                    },
+                    canCreateSetupIntents = true,
+                ),
+                intentConfirmationInterceptor = intentConfirmationInterceptor,
+            )
+
+            viewModel.handleViewAction(CustomerSheetViewAction.OnPrimaryButtonPressed)
+
+            val call = intentConfirmationInterceptor.calls.awaitItem()
+
+            assertThat(call).isEqualTo(
+                FakeIntentConfirmationInterceptor.InterceptCall.WithExistingPaymentMethod(
+                    initializationMode = PaymentSheet.InitializationMode.SetupIntent(
+                        clientSecret = "seti_123"
+                    ),
+                    paymentMethod = CARD_PAYMENT_METHOD,
+                    shippingValues = null,
+                    paymentMethodOptionsParams = null,
+                )
+            )
+
+            intentConfirmationInterceptor.calls.ensureAllEventsConsumed()
+        }
```

Comment:
`CustomerSheet` only uses `IntentConfirmationInterceptor` in this way. Makes no use of `shippingValues` or `paymentMethodOptionParams`